server/fd/io/feed_parsers/escenic_xmli.py

```diff
@@ -38,55 +40,109 @@ def can_parse(self, xml):
         return xml.tag == 'NewsML'
 
     def parse(self, xml, provider=None):
-        items = {}
+        items = {'associations': {}}
         try:
             self.parse_newslines(items, xml)
-            self.parse_media(items, xml)
+            self.parse_feature_media(items, xml)
             self.parse_news_identifier(items, xml)
             self.parse_metadata(items, xml)
             self.parse_byline(items, xml)
             self.parse_news_management(items, xml)
-
-            items['body_html'] = etree.tostring(
-                xml.find('NewsItem/NewsComponent/ContentItem/DataContent/nitf/body/body.content'),
-                encoding='unicode').replace('<body.content>', '').replace('</body.content>', '')
+            self.parse_body_html(items, xml)
 
             return items
         except Exception as ex:
             raise ParserError.newsmlTwoParserError(ex, provider)
 
-    # TODO check internal links like /12345656 (escenic id) and look for <NewsItemId>229126224</NewsItemId> in source_id
-    # relative links to category pages /sport/football and absolute links remain untouched
-
-    def parse_media(self, items, tree):
-        parsed_media = self.media_parser(
-            tree.findall('NewsItem/NewsComponent/ContentItem/DataContent/nitf/body/body.content/media/'))
-        items['associations'] = {
-            'featuremedia': {
-                'type': 'picture',
-                'guid': generate_tag_from_url(parsed_media[0]['source']),
-                'headline': items['headline'],
-                'creditline': parsed_media[0]['copyright'],
-                'description_text': parsed_media[0]['alternate-text'],
-                # 'firstcreated': items['versioncreated'],
-                # 'versioncreated': items['versioncreated'],
-                'renditions': {
-                    'baseImage': {
-                        'href': parsed_media[0]['source'],
-                        'width': parsed_media[0]['width'],
-                        'height': parsed_media[0]['height'],
-                        'mimetype': 'image/jpeg',
-                    },
-                    'viewImage': {
-                        'href': parsed_media[0]['source'],
-                        'width': parsed_media[0]['width'],
-                        'height': parsed_media[0]['height'],
-                        'mimetype': 'image/jpeg',
-                    }
+    def import_images(self, associations, name, attributes):
+        """ import images to mongo """
+        # if len(name) == 0 or len(attributes) == 0 or len(associations) == 0:
+        #     pass
+        # else:
+        associations[name] = {
+            'type': 'picture',
+            'guid': generate_tag_from_url(
+                attributes.get('source', '')),
+            'headline': '',
+            'creditline': '',
```

Comment:
i think we should keep the copyright info

Reply:
they are there now @jhalitschke